luigi/contrib/dataproc.py

```diff
@@ -0,0 +1,208 @@
+
+    def _get_cluster_status(self):
+        return self.dataproc_client.projects().regions().clusters()\
+            .get(projectId=self.gcloud_project_id, region=self.dataproc_region, clusterName=self.dataproc_cluster_name)\
+            .execute()
+
+    def complete(self):
+        try:
+            self._get_cluster_status()
+            return True  # No (404) error so the cluster already exists
+        except HttpError as e:
+            if e.resp.status == 404:
+                return False  # We got a 404 so the cluster doesn't exist yet
+            else:
+                raise e  # Something's wrong ...
+
+    def run(self):
+        base_uri = "https://www.googleapis.com/compute/v1/projects/{}".format(self.gcloud_project_id)
+        cluster_conf = {
+            "clusterName": self.dataproc_cluster_name,
+            "projectId": self.gcloud_project_id,
+            "config": {
+                "configBucket": "",
+                "gceClusterConfig": {
+                    "networkUri": base_uri + "/global/networks/" + self.gcloud_network,
+                    "zoneUri": base_uri + "/zones/" + self.gcloud_zone,
+                    "serviceAccountScopes": [
+                        "https://www.googleapis.com/auth/cloud-platform"
+                    ]
+                },
+                "masterConfig": {
+                    "numInstances": 1,
+                    "machineTypeUri": base_uri + "/zones/" + self.gcloud_zone + "/machineTypes/" + self.master_node_type,
+                    "diskConfig": {
+                        "bootDiskSizeGb": self.master_disk_size,
+                        "numLocalSsds": 0
+                    }
+                },
+                "workerConfig": {
+                    "numInstances": self.worker_normal_count,
+                    "machineTypeUri": base_uri + "/zones/" + self.gcloud_zone + "/machineTypes/" + self.worker_node_type,
+                    "diskConfig": {
+                        "bootDiskSizeGb": self.worker_disk_size,
+                        "numLocalSsds": 0
+                    }
+                },
+                "secondaryWorkerConfig": {
+                    "numInstances": self.worker_preemptible_count,
+                    "isPreemptible": True
+                }
+            }
+        }
+
+        self.dataproc_client.projects().regions().clusters()\
+            .create(projectId=self.gcloud_project_id, region=self.dataproc_region, body=cluster_conf).execute()
+
+        while True:
+            time.sleep(10)
+            cluster_status = self._get_cluster_status()
+            status = cluster_status['status']['state']
+            logger.info("Creating new dataproc cluster: {} status: {}".format(self.dataproc_cluster_name, status))
+            if status == 'RUNNING':
+                break
+            if status == 'ERROR':
+                raise Exception(cluster_status['status']['details'])
+
+
+class DeleteDataprocClusterTask(_DataprocBaseTask):
```

Comment:
You can extend the delete task to require a DataprocTask that does your calculation, then have that task require a cluster creation task.

What ends up running is: CreateCluster -> DataprocTask (do work on said cluster) -> Delete cluster.

However Since your dataproc task will need to be implemented to contain your own job that you actually want to run on the cluster I can't hard code that type of dependency chain in here.